internal/pkg/symnotify/symnotify_test.go

```diff
@@ -0,0 +1,140 @@
+package symnotify_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/openshift/cluster-logging-operator/internal/pkg/symnotify"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var Join = filepath.Join
+
+type Fixture struct {
+	T                   *testing.T
+	Root, Logs, Targets string
+	Watcher             *symnotify.Watcher
+}
+
+func NewFixture(t *testing.T) *Fixture {
+	t.Helper()
+	f := &Fixture{T: t}
+
+	var err error
+	f.Root, err = ioutil.TempDir("", t.Name())
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = os.RemoveAll(f.Root) })
+
+	f.Logs = Join(f.Root, "logs")
+	f.Targets = Join(f.Root, "targets")
+	for _, dir := range []string{f.Logs, f.Targets} {
+		require.NoError(t, os.Mkdir(dir, os.ModePerm))
+	}
+	f.Watcher, err = symnotify.NewWatcher()
+	require.NoError(t, err)
+	t.Cleanup(func() { f.Watcher.Close() })
+	return f
+}
+
+func (f *Fixture) Create(name string) (string, *os.File) {
+	f.T.Helper()
+	file, err := os.Create(name)
+	require.NoError(f.T, err)
+	f.T.Cleanup(func() { _ = file.Close() })
+	return name, file
+}
+
+func (f *Fixture) Link(name string) (string, *os.File) {
+	f.T.Helper()
+	target, file := f.Create(Join(f.Targets, name))
+	link := Join(f.Logs, name)
+	require.NoError(f.T, os.Symlink(target, link))
+	return link, file
+}
+
+func (f *Fixture) Event() symnotify.Event {
+	f.T.Helper()
+	e, err := f.Watcher.EventTimeout(time.Second)
+	require.NoError(f.T, err)
+	return e
+}
+
+func TestWatchesRealFiles(t *testing.T) {
+	f := NewFixture(t)
+	assert, require := assert.New(t), require.New(t)
```

Comment:
Can this be moved into the fixture to reduce code dup

Reply:
@alanconway seeking your advice here.